src/main/java/com/adobe/target/edge/client/service/DefaultTargetService.java

```diff
@@ -53,6 +55,18 @@ public DefaultTargetService(ClientConfig clientConfig, TelemetryService telemetr
     } else {
       this.targetHttpClient = targetHttpClient;
     }
+
+    Properties defaultProps = new Properties();
+    try {
+      InputStream in = getClass().getResourceAsStream("/gradle.properties");
+      BufferedReader reader = new BufferedReader(new InputStreamReader(in));
+      defaultProps.load(reader);
+      in.close();
+    } catch (IOException e) {
+      System.out.println("unable to location gradle.properties file");
+    }
+
+    this.SDK_VERSION = defaultProps.getProperty("version");
     this.targetHttpClient.addDefaultHeader(SDK_USER_KEY, SDK_USER_VALUE);
     this.targetHttpClient.addDefaultHeader(SDK_VERSION_KEY, SDK_VERSION);
```

Comment:
We probably want a null check for sdkVersion around this statement - just in case we hit that IOException above.  Otherwise we're potentially sending a header with a null value.

Reply:
What is the desired behavior if sdkVersion is null? Throw an an exception? As we are already notifying the client that their sdk version is undetermined